zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -1005,42 +1000,47 @@ private static List<String> applyUserSortField(
     return validationErrors;
   }
 
-  private static List<VariableValueFilter> toVariableValueFilters(
+  private static List<VariableValueFilter> toVariableValueFiltersForUserTask(
       final List<UserTaskVariableFilterRequest> filters) {
-
     if (filters != null && !filters.isEmpty()) {
-      return filters.stream()
-          .map(
-              filter -> {
-                final var builder = new VariableValueFilter.Builder().name(filter.getName());
-                if (filter.getValue() != null) {
-                  builder.valueOperation(UntypedOperation.of(eq(filter.getValue())));
-                }
-                return builder.build();
-              })
-          .toList();
+      final List<VariableValueFilter> variableValueFilters = new ArrayList<>();
+      filters.forEach(
+          filter ->
+              variableValueFilters.addAll(
+                  toVariableValueFilters(filter.getName(), filter.getValue())));
+      return variableValueFilters;
     }
-
     return null;
   }
 
   private static List<VariableValueFilter> toVariableValueFiltersForProcessInstance(
       final List<ProcessInstanceVariableFilterRequest> filters) {
-
     if (filters != null && !filters.isEmpty()) {
-      return filters.stream()
-          .map(
-              filter ->
-                  new VariableValueFilter.Builder()
-                      .name(filter.getName())
-                      .valueOperation(UntypedOperation.of(eq(filter.getValue())))
-                      .build())
-          .toList();
+      final List<VariableValueFilter> variableValueFilters = new ArrayList<>();
+      filters.forEach(
+          filter ->
+              variableValueFilters.addAll(
+                  toVariableValueFilters(filter.getName(), filter.getValue())));
+      return variableValueFilters;
     }
-
     return null;
   }
 
+  private static List<VariableValueFilter> toVariableValueFilters(
+      String name, StringFilterProperty value) {
+    Objects.requireNonNull(value);
```

Comment:
🔧 We should use the validation mechanism we usually apply in the request mapper. If this is null, it should go into a `List<String>` of violations, and the method should return a `Left<List<String>>`. This requires some changes in the mapper class since we don't have any `to...Filter` methods so far that can return an `Either`. So this is a new case.

💭 If we do it this way, I believe we're simply running into an Exception with this line and have an inconsistent API response. How about we add some test cases to the REST gateway to verify what happens in such cases:
* No `value` attribute provided in one variable filter object
* No `value` attribute provided in multiple variable filter objects
* A `value` attribute with a `null` value provided in one variable filter object